cs-office-365/src/main/java/io/cloudslang/content/office365/utils/PopulateMessageBody.java

```diff
@@ -0,0 +1,116 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudslang.content.office365.utils;
+
+import com.google.gson.Gson;
+import io.cloudslang.content.office365.entities.CreateMessageInputs;
+import io.cloudslang.content.office365.entities.Office365CommonInputs;
+import io.cloudslang.content.office365.entities.createMessageModels.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.cloudslang.content.office365.utils.Constants.CONTENT_TYPE;
+
+public class PopulateMessageBody {
+
+    public static String populateMessageBody(Office365CommonInputs commonInputs, CreateMessageInputs createMessageInputs, String delimiter) {
+
+        CreateMessageBody messageBody = new CreateMessageBody();
+        List<String> categoriesList = new ArrayList<>();
+        List<BccRecipient> bccList = new ArrayList<>();
+        List<CcRecipient> ccList = new ArrayList<>();
+        List<ToRecipient> toList = new ArrayList<>();
+        List<ReplyTo> replyToList = new ArrayList<>();
+        Body body = new Body();
+        From fromVal = new From();
+        Sender senderVal = new Sender();
+        ToRecipient toRecipient = new ToRecipient();
+        EmailAddress toEmailAddress = new EmailAddress();
+        EmailAddress senderEmailAddress = new EmailAddress();
+        EmailAddress emailAddress = new EmailAddress();
+        EmailAddress bccemailAddress = new EmailAddress();
+        EmailAddress ccemailAddress = new EmailAddress();
+        EmailAddress replyEmailAddress = new EmailAddress();
+        BccRecipient bccRecipient = new BccRecipient();
+        CcRecipient ccRecipient = new CcRecipient();
+        ReplyTo replyTo = new ReplyTo();
+
+
+        messageBody.setImportance(createMessageInputs.getImportance());
+        messageBody.setInferenceClassification(createMessageInputs.getInferenceClassification());
+        messageBody.setInternetMessageId(createMessageInputs.getInternetMessageId());
+        messageBody.setIsRead(Boolean.valueOf(createMessageInputs.getIsRead()));
+        messageBody.setIsDeliveryReceiptRequested(Boolean.valueOf(createMessageInputs.getIsDeliveryReceiptRequested()));
+        messageBody.setIsReadReceiptRequested(Boolean.valueOf(createMessageInputs.getIsReadReceiptRequested()));
+        messageBody.setSubject(createMessageInputs.getSubject());
+        messageBody.setId(commonInputs.getUserId());
+
+
+        String[] categoryList = createMessageInputs.getCategories().split(delimiter);
+        for (String category : categoryList) {
+            categoriesList.add(category);
+        }
+        messageBody.setCategories(categoriesList);
+
+        String[] list = createMessageInputs.getBccRecipients().split(delimiter);
+        for (String item : list) {
+            bccemailAddress.setAddress(item);
```

Comment:
This iteration is used 3 times in this method. Please add a new method with the same behavior and the according arguments so we avoid similar written more times.